File: api/src/Entity/ConferenceReference.php
<?php

namespace Crypter\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * @ORM\Entity(repositoryClass="Crypter\Repository\ConferenceReferenceRepository")
 */
class ConferenceReference
{
    /**
     * @ORM\Id()
     * @ORM\Column(type="guid")
     * @ORM\GeneratedValue(strategy="UUID")
     *
     * @Assert\Uuid()
     */
    private $uuid;

    /**
     * @ORM\ManyToOne(targetEntity="Crypter\Entity\User")
     * @ORM\JoinColumn(name="`user`", referencedColumnName="uuid", nullable=false)
     */
    private $user;

    /**
     * @ORM\ManyToOne(targetEntity="Crypter\Entity\Conference")
     * @ORM\JoinColumn(name="conference", referencedColumnName="uuid", nullable=false)
     */
    private $conference;

    /**
     * @ORM\Column(name="messages_count", type="integer", options={"default": 0})
     */
    private $messagesCount = 0;

    /**
     * @ORM\Column(name="unread_messages_count", type="integer", options={"default": 0})
     */
    private $unreadMessagesCount = 0;

    /**
     * @ORM\ManyToOne(targetEntity="Crypter\Entity\User")
     * @ORM\JoinColumn(name="participant", referencedColumnName="uuid", nullable=true)
     */
    private $participant;

    /**
     * @ORM\Column(name="updated_at", type="datetimetz")
     */
    private $updatedAt;

    /**
     * @ORM\ManyToOne(targetEntity="Crypter\Entity\Message")
     * @ORM\JoinColumn(name="last_message", referencedColumnName="uuid", nullable=false)
     */
    private $lastMessage;
    
    public function __construct()
    {
        $this->updatedAt = (new \DateTime());
    }

    public function getUuid(): ?string
    {
        return $this->uuid;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): self
    {
        $this->user = $user;

        return $this;
    }

    public function getConference(): ?Conference
    {
        return $this->conference;
    }

    public function setConference(?Conference $conference): self
    {
        $this->conference = $conference;

        return $this;
    }

    public function getMessagesCount(): ?int
    {
        return $this->messagesCount;
    }

    public function setMessagesCount(int $messagesCount): self
    {
        $this->messagesCount = $messagesCount;

        return $this;
    }

    public function getUnreadMessagesCount(): ?int
    {
        return $this->unreadMessagesCount;
    }

    public function setUnreadMessagesCount(int $unreadMessagesCount): self
    {
        $this->unreadMessagesCount = $unreadMessagesCount;

        return $this;
    }

    public function getParticipant(): ?User
    {
        return $this->participant;
    }

    public function setParticipant(?User $participant): self
    {
        $this->participant = $participant;

        return $this;
    }

    public function getUpdatedAt(): ?\DateTimeInterface
    {
        return $this->updatedAt;
    }

    public function setUpdatedAt(\DateTimeInterface $updatedAt): self
    {
        $this->updatedAt = $updatedAt;

        return $this;
    }

    public function getLastMessage(): ?Message
    {
        return $this->lastMessage;
    }

    public function setLastMessage(?Message $lastMessage): self
    {
        $this->lastMessage = $lastMessage;

        return $this;
    }
}

File: ws/src/models/conference_reference.py
import uuid
import datetime

from sqlalchemy import Column, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.dialects.postgresql import UUID, INTEGER, TIMESTAMP

from database import Base

# from user import User
# from conference import Conference

class Conference_Reference(Base):
    __tablename__ = 'conference_reference'

    uuid = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_uuid = Column('user', UUID(as_uuid=True), ForeignKey('user.uuid'), nullable=False)
    conference_uuid = Column('conference', UUID(as_uuid=True), ForeignKey('conference.uuid'), nullable=False)
    messages_count = Column(INTEGER, nullable=False, default=0)
    unread_messages_count = Column(INTEGER, nullable=False, default=0)
    participant_uuid = Column('participant', UUID(as_uuid=True), ForeignKey('user.uuid'))
    updated_at =  Column(TIMESTAMP(timezone=True), nullable=False, default=datetime.datetime.utcnow)
    last_message_uuid = Column('last_message', UUID(as_uuid=True), ForeignKey('message.uuid'))

    user = relationship("User", foreign_keys=[user_uuid])
    conference = relationship("Conference")
    participant = relationship("User", foreign_keys=[participant_uuid])
    last_message = relationship("Message", foreign_keys=[last_message_uuid])
